flutter/shell/platform/tizen/flutter_tizen_view.h

```diff
@@ -112,6 +113,35 @@ class FlutterTizenView : public TizenViewEventHandlerDelegate {
   TextInputChannel* text_input_channel() { return text_input_channel_.get(); }
 
  private:
+  // Struct holding the state of an individual pointer. The engine doesn't keep
+  // track of which buttons have been pressed, so it's the embedder's
+  // responsibility.
+  struct PointerState {
+    // The device kind.
+    FlutterPointerDeviceKind device_kind = kFlutterPointerDeviceKindTouch;
+
+    // A virtual pointer ID that is unique across all device kinds.
+    int32_t pointer_id = 0;
+
+    // True if the last event sent to Flutter had at least one button pressed.
+    bool flutter_state_is_down = false;
+
+    // True if kAdd has been sent to Flutter. Used to determine whether
+    // to send a kAdd event before sending an incoming pointer event, since
+    // Flutter expects pointers to be added before events are sent for them.
+    bool flutter_state_is_added = false;
```

Comment:
I think flutter_state_* is redundant. How about `is_down`, `is_added`, not flutter_state_*?

Reply:
The design is from the [Windows embedder](https://github.com/flutter/engine/blob/main/shell/platform/windows/flutter_windows_view.h).

I'd keep using the `flutter_state_` prefix since it makes the meaning more explicit and clearer. Actually `is_down` has slightly different meanings for Flutter and the embedder. For Flutter, as you can see in the `FlutterPointerPhase` documentation, the down state means that _any_ (one or more) of the mouse buttons are pressed. In the embedder, however, we track the state of each mouse button individually, so if you just say `is_down`, I have to ask "which button is pressed?"